### ISSUE TITLE
MOOSE Newsletter

### ISSUE THREAD

**Issue by aeslaughter** [idaholab/moose#11446]
## Rationale
Users and management require occasional updates on changes to MOOSE.

## Description
Each month we will release a "newsletter" that discuss the major changes to MOOSE that occurred and what is planned for the coming months. The content for the newsletter will be posted on our website so we have a complete record, it will also be emailed to the moose-users list once it is merged.

## Impact
Our users and management will be happier.

**Comment by WilkAndy** [comment]
A related aside: i think you guys must have got the moosedocs system pretty-much right for users like me.  I just had a look at the Issues and of the 7 most recent Issues, 4 of them are about adding/modifying documentation!   Good news for MOOSE usability.

**Comment by friedmud** [comment]
@WilkAndy yep... @aeslaughter built something great... and we're super motivated to fill it in.

**Comment by permcody** [comment]
This issue doesn't need to be open. We can reference it when closed.

**Comment by hugary1995** [comment]
@permcody @MengnanLi91 can we not automate this with a bot? An example workflow:
1. Use GHAPI to fetch all commits messages, issue and PR descriptions in the past month.
2. Use a LM to write the news given a template.
3. Create a PR. Tag all contributors.
4. Merge PR.

Then run this workflow monthly.

**Comment by MengnanLi91** [comment]
I think that's a great idea! Actually do you know Dosu? It is a github helper agent that does auto reply, newsletter PR and more. Langchain and LlamaIndex all use it for their repo, but it requires a subscription fee for code like MOOSE.
We can definitely write our own agent, but currently the LLM hosting part is the roadblock for us. We will need a server machine to host the LLM(the GitHub workflow free version VM is too small for LLM) and run it constantly. We need to talk our HPC team to figure out how to host it. We are exploring that option with MOOSE discussion bot RANGER.
Maybe that's something AWS can help, what do you think @permcody

**Comment by hugary1995** [comment]
This is the first time I heard Dosu. I just briefly browsed their homepage. I don't see much value it would bring to us, especially we can achieve this workflow with a few python scripts. If it's free then for sure, but it sounds like it is not.

I thought INL already has a LLM hosted on the intranet. I might have a false memory. Or perhaps the issue lies in that CIVET can't access it?

**Comment by MengnanLi91** [comment]
> This is the first time I heard Dosu. I just briefly browsed their homepage. I don't see much value it would bring to us, especially we can achieve this workflow with a few python scripts. If it's free then for sure, but it sounds like it is not.
> 
> I thought INL already has a LLM hosted on the intranet. I might have a false memory. Or perhaps the issue lies in that CIVET can't access it?

Not sure which LLM you refer to? We have some internal tools but I don't remember we have any in the public domain.

If we only keep it backend, run it manually every month. I can see that be done pretty quickly.

Maybe we could just leverage ChatGPT codex for that.

**Comment by dschwen** [comment]
@MengnanLi91 we could use the locally hosted gemini 120B for that. All we need is a server (or vm) that can access the internet (i.e. github API). We already have that, too.

**Comment by MengnanLi91** [comment]
> @MengnanLi91 we could use the locally hosted gemini 120B for that. All we need is a server (or vm) that can access the internet (i.e. github API). We already have that, too.

@loganharbour and I has discussed using MOOSE training machine to deploy RANGER. I think this can also be deployed there. However, I think we can just use codex with agent mode to do this, all we need is to write a prompt template for the task. I think we can create a customized GPT using codex and write what we need as system prompt. OpenAI also have scheduler so that we can do it every month. This is all included in INL GPT enterprise. I can give it a try next week and see how good it is

**Comment by hugary1995** [comment]
Yeah, what you described above is the best solution I can think of.

**Comment by MengnanLi91** [comment]
@cticenhour

**Comment by cticenhour** [comment]
Thank you for looping me in @MengnanLi91. I would be supportive of automating some aspects of this, as it would make my job of getting the news reviewed and pushed through much easier by shortening the collection / input step. Mengnan just sent me a sample instruction set for a GPT that she created to explore these ideas; I'll give some feedback and work with her on it. Unfortunately I don't (yet) have an INL license to ChatGPT.

Automating this could also serve as a low-risk test bed for how we seriously integrate AI input into our formal SQA process, which is also something I've been thinking about a lot lately.

**Comment by hugary1995** [comment]
> Automating this could also serve as a low-risk test bed for how we seriously integrate AI input into our formal SQA process, which is also something I've been thinking about a lot lately.

Ha, I also thought about that. I think that could be very useful as well. I think the key difference is _where_ the AI integration takes place. For generating the newsletter, we only run the language model on server side, like civet. For writing SQA docs, like `requirement = ` in test specs, we will have to run a fine-tuned language model at client side, e.g., run by the developers who are writing the test cases. Distributing/hosting the fine-tuned language model could be challenging. So I agree with you -- starting with the newsletter is a nice first-step low-risk test bed. Let's see what happens!

Feel free to open a separate issue on that subject and loop in more people.
